contrib/server/operations/pbench-upload-results.py

```diff
@@ -135,21 +132,29 @@ def main() -> int:
     if parsed.metadata:
         metadata.extend(parsed.metadata)
 
+    # Process date range filtering arguments
     since: Optional[datetime.datetime] = None
     before: Optional[datetime.datetime] = None
 
     since_ts: Optional[float] = None
     before_ts: Optional[float] = None
 
     if parsed.since:
-        since = date_parser.parse(parsed.since)
+        since = dateutil.parser.parse(parsed.since)
         since_ts = since.timestamp()
 
     if parsed.before:
-        before = date_parser.parse(parsed.before)
+        before = dateutil.parser.parse(parsed.before)
         before_ts = before.timestamp()
 
     if since and before:
+        if before <= since:
+            print(
+                f"SINCE ({since}) must be earlier than BEFORE ({before})",
+                file=sys.stderr,
+            )
```

Comment:
This is very slightly annoying:  if you want to upload one day's worth, then you have to provide different values for `before` and `since`.  (I understand that that _is_ the design, here, but it strikes me as a usability pothole -- I think it would be "more usable" if the range were inclusive on both ends.)

Reply:
I really don't see any point in bothering with "smart roundoff" for such a special purpose admin utility. What would be the point?